### PR TITLE
Vehicle: update HealthAndArmingCheck

### PR DESCRIPTION
**Pull request by bkueng**
From https://github.com/mavlink/libevents/pull/3

### CODE REVIEW

File: src/Vehicle/EventHandler.h [hunk 1]
```diff
@@ -56,6 +57,7 @@ class EventHandler : public QObject
     QTimer _timer;
     events::parser::Parser _parser;
     events::HealthAndArmingChecks _healthAndArmingChecks;
+    bool _healthAndArmingChecksValid{false};
```
Comment (patrickelectric) [inline src/Vehicle/EventHandler.h:60]: The variable value should be set on constructor initializer list, as the current class style.

**Review by patrickelectric — CHANGES_REQUESTED** [review]
The update on libevents could be done in a separated commit (and in preference in a separated PR), an libevents update is necessary for this changes ?

**Comment by bkueng** [comment]
> The update on libevents could be done in a separated commit (and in preference in a separated PR), an libevents update is necessary for this changes ?

Yes it's an API change. I separated the commits

File: src/Vehicle/EventHandler.h (hunk 1 above)
Comment (bkueng) [reply src/Vehicle/EventHandler.h:60]: It is consistent, only members initialized from constructor arguments use the initializer list.

**Comment by bkueng** [comment]
Merging as it's needed for PX4. @patrickelectric let me know if you want to discuss something further.

**Pull request merged**